models/base_model.py

```diff
@@ -48,9 +48,14 @@ def update(cls, schema, id, payload):
         return obj
 
     @staticmethod
-    def validate(schema, payload, partial=False):
+    def validate(schema, payload, context=None, partial=False):
         try:
-            return schema().load(payload, unknown=EXCLUDE, partial=partial)
+            if context:
+                return schema(context=context).load(
+                    payload, unknown=EXCLUDE, partial=partial
+                )
+            else:
+                return schema().load(payload, unknown=EXCLUDE, partial=partial)
```

Comment:
I wonder if we can get rid of this logic here and simply do this: 
`return schema(context=context).load(payload, unknown=EXCLUDE, partial=partial)`